src/main/java/com/internet/shop/controllers/AddProductController.java

```diff
@@ -0,0 +1,35 @@
+package com.internet.shop.controllers;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import java.math.BigDecimal;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
```

Comment:
Let's make all of the service objects not final

Reply:
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/homework/lesson_16_hw.html
In this example service is final. So, I guess it should be final :)